src/sentry/api/endpoints/sentry_app_authorizations.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry.api.bases import SentryAppAuthorizationEndpoint as BaseEndpoint
+from sentry.api.authentication import ClientIdSecretAuthentication
+from sentry.coreapi import APIUnauthorized
+from sentry.mediators.sentry_app_installations import Authorizer
+from sentry.api.serializers.models.apitoken import ApiTokenSerializer
+
+
+class SentryAppAuthorizationsEndpoint(BaseEndpoint):
+    authentication_classes = (ClientIdSecretAuthentication, )
+
+    def post(self, request, install):
+        try:
+            token = Authorizer.run(
+                grant_type=request.json_body.get('grant_type'),
+                code=request.json_body.get('code'),
+                client_id=request.json_body.get('client_id'),
+                user=request.user,
+                install=install,
```

Comment:
Many OAuth2 implementations also provide a `state` property which prevents CSRF/replays. We might want to do that as well.

Reply:
Added.